readthedocs/oauth/notifications.py

```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+from __future__ import division, print_function, unicode_literals
+
+from django.utils.translation import ugettext_lazy as _
+from readthedocs.notifications import SiteNotification
+
+
+class AttachWebhookNotification(SiteNotification):
+
+    NO_CONNECTED_SERVICES = 'no_connected_services'
+    NO_PERMISSIONS = 'no_permissions'
+    NO_ACCOUNTS = 'no_accounts'
+
+    context_object_name = 'provider'
+    success_message = _('Webhook activated successfully.')
+    failure_message = {
+        NO_CONNECTED_SERVICES: _('Webhook activation failed. There are no connected services for ''this project.'),  # noqa
+        NO_PERMISSIONS: _('Webhook activation failed. Make sure you have permissions to set it.'),  # noqa
+        NO_ACCOUNTS: _('No accounts available to set webhook on. Please connect your {provider.name} account.'),  # noqa
```

Comment:
For the `success_message` and `failure_message`, the copy needs to be unified a bit.

* I'd use `add` instead of `activate`, as the webhook doesn't exist before we add it
* Similarly, `set webhook` should be `add webhook`
* I'd still give a link to the user to rectify the problem even though these aren't persistent messages.

So, for example:

```python
success_message = _('Webhook successfully added')
failure_message = {
    NO_CONNECTED_SERVICES: _('Could not add webhook for {project.name}. Please <a href="{url}">connect your {provider} account</a> first.'),
    NO_PERMISSIONS: _('Could not add webhook for {project.name}. Make sure <a href="{url}">you have the correct {provider.name} permissions</a>.'),
    NO_ACCOUNTS: _('Could not add webhook for {project.name}. Please <a href="{url}">connect your {provider.name} account</a>.'),
}
```

Also, it looks like NO_CONNECTED_SERVICES and NO_ACCOUNTS are effectively the same thing from the user's perspective?

Reply:
Mhm, `NO_CONNECTED_SERVICES` is shown when there is no services are registered in our registry --it's empty (https://github.com/rtfd/readthedocs.org/blob/master/readthedocs/oauth/services/__init__.py#L26). It shouldn't be a notification _to_ the user, in fact. Maybe that was added for debug reasons and can be completely removed?

Reply:
In your example, you are using `provider` and `project` as context variables. At the moment, there is only one context variable supported. I will adapt it to support multiple objects as context.